app/actions/client.py

```diff
@@ -0,0 +1,725 @@
+            f"are missing. Please fix the integration setup in the portal."
+        )
+    return AuthenticateConfig.parse_obj(auth_config.data)
+
+
+async def get_token(integration, config):
+    current_state = await state_manager.get_state(
+        str(integration.id),
+        "auth",
+        config.email
+    )
+
+    if current_state:
+        return current_state["token"]
+
+    url = f"{integration.base_url}/auth/login"
+    async with httpx.AsyncClient(timeout=120) as session:
+        response = await session.post(
+            url=url,
+            json={
+                "email": config.email,
+                "password": config.password
+            }
+        )
+        response.raise_for_status()
+
+    state = {
+        "token": response.json()["data"]
+    }
+    await state_manager.set_state(
+        str(integration.id),
+        "auth",
+        state,
+        config.email,
+    )
+
+    return response.json()["data"]
+
+
+async def get_aoi_data(integration, config):
+    auth = get_auth_config(integration)
+    try:
+        token = await get_token(integration, auth)
+        aoi_id = aoi_from_url(config.url)
+        aoi_data = await get_aoi(integration, auth, aoi_id, token)
+    except Exception as e:
+        message = f"Unhandled exception occurred. Exception: {e}"
+        logger.exception(
+            message,
+            extra={
+                "integration_id": str(integration.id),
+                "attention_needed": True
+            }
+        )
+        raise e
+    else:
+        return aoi_data
+
+
+async def get_fire_alerts(aoi_data, integration, config):
+    auth = get_auth_config(integration)
+
+    logger.info(
+        f"Processing fire alerts for '{auth.email}' ",
+        extra={
+            "integration_id": str(integration.id),
+            "endpoint": integration.base_url,
+            "username": auth.email,
+        },
+    )
+
+    try:
+        end_date = datetime.utcnow()
+        start_date = end_date - timedelta(days=DEFAULT_FIRE_LOOKBACK_DAYS)
+
+        alerts = await get_fire_alerts_response(integration, config, aoi_data, start_date, end_date)
+
+    except pydantic.ValidationError as ve:
+        message = f'Error while parsing GFW "get_fire_alerts" endpoint. {ve.json()}'
+        logger.exception(
+            message,
+            extra={
+                "integration_id": str(integration.id),
+                "attention_needed": True
+            }
+        )
+        raise ve
+
+    except Exception as e:
+        message = f"Unhandled exception occurred. Exception: {e}"
+        logger.exception(
+            message,
+            extra={
+                "integration_id": str(integration.id),
+                "attention_needed": True
+            }
+        )
+        raise e
+    else:
+        logger.info(f"Got {len(alerts)} fire alerts")
+        return alerts
+
+
+async def get_glad_alerts(aoi_data, integration, config):
```

Comment:
These are no longer GLAD alert, but are instead "Integrated Alerts". 
Perhaps we can take this opportunity to update the names in code.